### PR TITLE
Loki: Fix error when some queries return zero results

### LINKED ISSUE #28621: Loki: Error when one query returns zero results
Seems like there is a weird error that happens only when running 2 queries and one of then does not return any data.

Screenshot from 2020-10-28 23-06-27
Screenshot from 2020-10-28 23-06-19

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Fixes Loki error when some queries return zero results. I have added tests and comments.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/28621

### CODE REVIEW

**Comment by torkelo** [comment]
I am not familiar with the logs model code so not sure I am the best to review this. maybe @aocenas or @kaydelaney are more familiar?

**Comment by ivanahuckova** [comment]
@torkelo those automatic assignments 🙈. Changed it to obs-squad people.

File: public/app/core/logs_model.test.ts [hunk 1]
```diff
@@ -700,6 +700,61 @@ describe('logSeriesToLogsModel', () => {
...
+        refId: 'A',
+        meta: {
+          searchWords: ['test'],
+          limit: 1000,
+          stats: [{ displayName: 'Summary: total bytes processed', value: 97048, unit: 'decbytes' }],
+          custom: { lokiQueryStatKey: 'Summary: total bytes processed' },
+          preferredVisualisationType: 'logs',
+        },
+      }),
+      {
```
Comment (aocenas) [inline public/app/core/logs_model.test.ts:737]: Why don't we need to use `toDataFrame` here? Was a bit confused and thought we area creating only one dataframe here.

File: public/app/core/logs_model.test.ts [hunk 2]
```diff
@@ -700,6 +700,61 @@ describe('logSeriesToLogsModel', () => {
...
+          limit: 1000,
+          stats: [{ displayName: 'Summary: total bytes processed', value: 97048, unit: 'decbytes' }],
+          custom: { lokiQueryStatKey: 'Summary: total bytes processed' },
+          preferredVisualisationType: 'logs',
+        },
+      },
+    ];
+
+    const logsModel = dataFrameToLogsModel(logSeries, 0, 'utc');
+    expect(logsModel.meta).toMatchObject([
```
Comment (aocenas) [inline public/app/core/logs_model.test.ts:752]: Can we also check if we actually processed the lines from the dataframe that has fields?

**Review by aocenas — COMMENTED** [review]
Good work, left some small comments.

File: public/app/core/logs_model.test.ts (hunk 1 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.test.ts]: yes, done in https://github.com/grafana/grafana/pull/28645/commits/a0b9ed162bc36a9a13cb14c3a3d93564dbaf582f

File: public/app/core/logs_model.test.ts (hunk 2 above)
Comment (ivanahuckova) [reply public/app/core/logs_model.test.ts:751]: yes, done in a0b9ed1

File: public/app/core/logs_model.ts [hunk 3]
```diff
@@ -306,8 +307,12 @@ export function logSeriesToLogsModel(logSeries: DataFrame[]): LogsModel | undefi
   // Find the fields we care about and collect all labels
   let allSeries: LogFields[] = [];
 
-  if (hasFields(logSeries)) {
-    allSeries = logSeries.map(series => {
+  // We are sometimes passing data frames with no fields because we want to calculate correct meta stats.
+  // Therefore we need to filer out series with no fields. These series are used only for meta stats calculation.
```
Comment (zoltanbedi) [inline public/app/core/logs_model.ts:311]:
```suggestion
  // Therefore we need to filter out series with no fields. These series are used only for meta stats calculation.
```

**Review by zoltanbedi — APPROVED** [review]
lgtm, just a small typo

**Pull request merged**